ament_cppcheck/ament_cppcheck/main.py

```diff
@@ -118,20 +122,20 @@ def main(argv=sys.argv[1:]):
         # the number of cores cannot be determined, do not extend args
         pass
 
-    # detect cppcheck 1.88 which caused issues
-    if 'AMENT_CPPCHECK_ALLOW_1_88' not in os.environ:
-        if cppcheck_version == '1.88':
+    # detect cppcheck 1.88 or 2.x which are much too slow
+    if 'AMENT_CPPCHECK_ALLOW_SLOW_VERSIONS' not in os.environ:
+        if cppcheck_version == '1.88' or cppcheck_version.startswith('2.'):
             print(
-                'cppcheck 1.88 has known performance issues and therefore will not be used, '
-                'set the AMENT_CPPCHECK_ALLOW_1_88 environment variable to override this.',
+                f'cppcheck {cppcheck_version} has known performance issues and therefore will not '
+                'be used, set the AMENT_CPPCHECK_ALLOW_1_88 environment variable to override this.',
```

Comment:
```suggestion
                'be used, set the AMENT_CPPCHECK_ALLOW_SLOW_VERSIONS environment variable to override this.',
```

Reply:
Bah, good catch.  Fixed in 58ad06d.  I tested this out locally and it seems to be happy.  I'm not going to run another full CI on this; once the previous CI is done I'll go ahead and merge.